util/gpool/spmc/spmcpool.go

```diff
@@ -260,9 +259,9 @@ func (p *Pool[T, U, C, CT, TF]) AddProduceBySlice(producer func() ([]T, error),
 	taskID := p.NewTaskID()
 	var wg sync.WaitGroup
 	result := make(chan U, opt.ResultChanLen)
-	closeCh := make(chan struct{})
+	productExitCh := make(chan struct{})
 	inputCh := make(chan pooltask.Task[T], opt.TaskChanLen)
```

Comment:
https://github.com/uber-go/guide/blob/master/style.md#channel-size-is-one-or-none

In my experience, it's really helpful to force us to think about more extreme situations and avoid bug in advance. I'm more like unbuffered channel now because we always need to deal with sender's blocking (golang has no unlimited length channel) and unbuffered channel means we do not need to drain elements when receiver is closing

Reply:
Agree with you. Let me set TaskChanLen's default value as 1.

Reply:
Done